pkg/util/gitutil/git.go

```
@@ -55,48 +70,85 @@ func GetGitRepository(dir string) (*git.Repository, error) {
 	project := ""
+	vcsKind := ""
 
-	const GitHubSSHPrefix = "git@github.com:"
-	const GitHubHTTPSPrefix = "https://github.com/"
-	const GitHubRepositorySuffix = ".git"
+	if cloudSourceControlSSHRegex.MatchString(remoteURL) {
+		// Get all matching groups.
+		matches := cloudSourceControlSSHRegex.FindAllStringSubmatch(remoteURL, -1)[0]
+		// Get the named groups in our regex.
+		groupNames := cloudSourceControlSSHRegex.SubexpNames()
 
-	if strings.HasPrefix(remoteURL, GitHubSSHPrefix) {
-		project = trimGitRemoteURL(remoteURL, GitHubSSHPrefix, GitHubRepositorySuffix)
-	} else if strings.HasPrefix(remoteURL, GitHubHTTPSPrefix) {
-		project = trimGitRemoteURL(remoteURL, GitHubHTTPSPrefix, GitHubRepositorySuffix)
+		groups := map[string]string{}
+		for i, value := range matches {
+			groups[groupNames[i]] = value
+		}
+		vcsKind = groups["host_name"]
+		project = groups["owner_and_repo"]
+		project = strings.TrimSuffix(project, defaultGitCloudRepositorySuffix)
+	} else {
+		if parsedURL, err := url.Parse(remoteURL); err == nil {
+			vcsKind = parsedURL.Host
+			project = parsedURL.Path
+			// Replace the .git extension from the path.
+			project = strings.TrimSuffix(project, defaultGitCloudRepositorySuffix)
+			// Remove the prefix "/". TrimPrefix returns the same value if there is no prefix.
+			// So it is safe to use it instead of doing any sort of substring matches.
+			project = strings.TrimPrefix(project, "/")
+		} else {
+			return "", "", "", errors.Wrapf(err, "detecting the VCS info from the HTTPS remote URL %v", remoteURL)
+		}
 	}
 
+	// At this point, project is either an empty string or contains the information that we want.
+	// It is OK to run the split on the empty string anyway.
 	split := strings.Split(project, "/")
 
 	if len(split) != 2 {
-		return "", "", errors.Errorf("could not detect GitHub project from url: %v", remote)
+		return "", "", "", errors.Errorf("could not detect VCS project from url: %v", remoteURL)
```

If the remote is using some unexpected format -- e.g. some VCS host we've never heard of before -- it shouldn't be an error if we cannot parse it. We just don't have any tags to return.

The VCS information is optional. We shouldn't fail the update if it isn't provided.

---

Even though this function returns an errors, the callers are already ignoring any errors during VCS detection.